openhands/sdk/llm/message.py

```diff
@@ -13,6 +13,76 @@
 logger = get_logger(__name__)
 
 
+class MessageToolCall(BaseModel):
+    """Transport-agnostic tool call representation.
+
+    One canonical id is used for linking across actions/observations and
+    for Responses function_call_output call_id.
+    """
+
+    id: str = Field(..., description="Canonical tool call id")
+    name: str = Field(..., description="Tool/function name")
+    arguments: str = Field(..., description="JSON string of arguments")
+    origin: Literal["completion", "responses"] = Field(
+        ..., description="Originating API family"
+    )
+    raw: ChatCompletionMessageToolCall | ResponseFunctionToolCall | None = Field(
+        default=None,
+        description="Original provider object for advanced consumers",
+        exclude=True,
+    )
+
+    def __contains__(self, key: str) -> bool:
+        """Support 'in' operator for dictionary-style access."""
+        return hasattr(self, key)
+
+    def get(self, key: str, default: Any = None) -> Any:
+        """Get attribute with default value, similar to dict.get()."""
+        return getattr(self, key, default)
+
+    def __getitem__(self, key: str) -> Any:
+        """Support dictionary-style access to attributes."""
+        return getattr(self, key)
+
+    def __setitem__(self, key: str, value: Any) -> None:
+        """Support dictionary-style assignment of attributes."""
+        setattr(self, key, value)
+
```

Comment:
```suggestion
    def __contains__(self, key: str) -> bool:
        """Support 'in' operator for dictionary-style access."""
        return hasattr(self, key)

    def get(self, key: str, default: Any = None) -> Any:
        """Get attribute with default value, similar to dict.get()."""
        return getattr(self, key, default)

    def __getitem__(self, key: str) -> Any:
        """Support dictionary-style access to attributes."""
        return getattr(self, key)

    def __setitem__(self, key: str, value: Any) -> None:
        """Support dictionary-style assignment of attributes."""
        setattr(self, key, value)

```

Seems like maybe we don't need these, it's a simple wrapper over one thing?

Reply:
@enyst deleted